spec/features/happy_journeys/js/when_choose_lead_mentor_course.rb

```diff
@@ -0,0 +1,193 @@
+      page.check("Yes, I agree my information can be shared", visible: :all)
+    end
+
+    allow(ApplicationSubmissionJob).to receive(:perform_later).with(anything)
+
+    expect_page_to_have(path: "/registration/check-answers", submit_button_text: "Submit", submit_form: true) do
+      expect_check_answers_page_to_have_answers(
+        {
+
+          "Course" => "NPQ for Leading Teacher Development (NPQLTD)",
+          "Employer" => "",
+          "Employment type" => "As a lead mentor for an accredited ITT provider",
+          "ITT Provider" => approved_itt_provider_legal_name,
+          "Lead provider" => "Church of England",
+          "Role" => "",
+          "What setting do you work in?" => "Other",
+          "Where do you work?" => "England",
+        },
+      )
+    end
+
+    expect_page_to_have(path: "/registration/confirmation", submit_form: false) do
+      expect(page).to have_text("Your initial registration is complete")
+    end
+
+    expect(User.count).to eql(1)
+
+    User.last.tap do |user|
+      expect(user.email).to eql("user@example.com")
+      expect(user.full_name).to eql("John Doe")
+      expect(user.trn).to eql("1234567")
+      expect(user.trn_verified).to be_truthy
+      expect(user.trn_auto_verified).to be_falsey
+      expect(user.date_of_birth).to eql(Date.new(1980, 12, 13))
+      expect(user.national_insurance_number).to eq(nil)
+      expect(user.applications.count).to eql(1)
+
+      user.applications.first.tap do |application|
+        expect(application.eligible_for_funding).to be_falsey
+        expect(application.work_setting).to eql("other")
+        expect(application.employment_type).to eql("lead_mentor_for_accredited_itt_provider")
+      end
+    end
+
+    navigate_to_page(path: "/account", axe_check: false, submit_form: false) do
+      expect(page).to have_text("Church of England")
+      expect(page).to have_text("NPQ for Leading Teacher Development (NPQLTD)")
+    end
+
+    visit "/registration/share-provider"
+
+    expect_page_to_have(path: "/", axe_check: false, submit_form: false) do
+      expect(page).to have_content("Before you start")
+    end
+
+    expect(retrieve_latest_application_user_data).to eq(
+      "active_alert" => nil,
+      "admin" => false,
+      "date_of_birth" => "1980-12-13",
+      "ecf_id" => nil,
+      "email" => "user@example.com",
+      "flipper_admin_access" => false,
+      "full_name" => "John Doe",
+      "national_insurance_number" => nil,
+      "otp_expires_at" => nil,
+      "otp_hash" => nil,
+      "provider" => "tra_openid_connect",
+      "raw_tra_provider_data" => stubbed_callback_response_as_json,
+      "trn" => "1234567",
+      "trn_auto_verified" => false,
+      "trn_verified" => true,
+      "uid" => user_uid,
+    )
+
+    expect(retrieve_latest_application_data).to eq(
+      "cohort" => 2022,
+      "course_id" => Course.find_by_code(code: :NPQLTD).id,
+      "ecf_id" => nil,
+      "eligible_for_funding" => false,
```

Comment:
Is this correct? The user picked NPQLTD and selected an approved ITT Provider. They should be funded shouldn't they? It seems based on the funding_eligiblity_status_code below that they were marked ineligible because the first line of the calculator kicked in. The ITT check should happen before the institution check.

Reply:
Same now changed from testing